_vector-search/ai-search/building-agentic-search-flows.md

```diff
@@ -25,27 +28,99 @@ To configure new agents, be sure to first provision appropriate models. For work
+Ensure that you have a sufficient number of documents in your cluster to reasonably evaluate your agentic searches.
 
-## Example: Product search with GPT-5
+## Accessing the plugin
+
+To access the plugin, go to **OpenSearch Dashboards** and select **OpenSearch Plugins** > **AI Search Flows** from the top menu.
+
+## Configuring agents
 
-This example uses a deployed OpenAI GPT-5 model described in [this documentation]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/agent-customization/#gpt-5-recommended).
+Agents are very customizable and can be configured in many different ways depending on your target use case. The following image shows a fully-configured conversational agent.
+
+![Agent configuration]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agent-configuration.png)
+
+### Agent types
+
+`Flow` agents are optimized for speed and simplicity in query generation. `Conversational` agents can be configured with more tools and functionality for deeper thinking. Agent types can only be configured **once** during agent creation.
+
+When configuring `Flow` agents, certain models may require manually adding an appropriate `response_filter` in the **Query Planning** tool. For more details, see [Register a flow agent]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/flow-agent/#step-4-register-a-flow-agent). Currently, only `OpenAI` and `Amazon Bedrock Converse` are supported. If you would like native support for other providers, please open a [GitHub issue](https://github.com/opensearch-project/dashboards-flow-framework/issues).
 {: .note}
 
-To build and test your agentic search workflow in OpenSearch Dashboards, follow these steps:
+### Models
+
+In `Conversational` agents, models are responsible for query generation and tool orchestration. In `Flow` agents, models are used strictly for query generation. Additionally, different models are typically optimized for different scenarios: cost-efficient, fast inference versus resource-intensive, deep-thinking approaches.
+
+For a list of suggested models compatible with agentic search, see [Model configuration]({{site.url}}{{site.baseurl}}/vector-search/ai-search/agentic-search/agent-customization/#model-configuration).
+
+### Tools
+
+Agents must have the **Query Planning** tool enabled for executing agentic searches. Query generation can be entirely LLM-generated (default), or can be steered to select from a list of available search templates. Search templates can be useful for maintaining control over the queries generated, forcing the model to leverage known, working, and performant queries.
+![Search templates]({{site.url}}{{site.baseurl}}/images/dashboards-flow-framework/agentic-search-search-template.png)
+
+Other pre-built tools are available for `Conversational` agents, including **Search Index**, **List Index**, **Index Mapping**, and **Web Search**. Enable these to extend your agent's capabilities.
```

Comment:
we can link the complete set of built in tools from opensearch here

Reply:
Listed the ones with first-class support on UI. Good point, will add a line to reference other existing tools
```
While these represent the most commonly used tools in agentic search, several additional [OpenSearch tools are available]({{site.url}}{{site.baseurl}}/ml-commons-plugin/agents-tools/tools/index/) and can be configured manually using the JSON view.

```